core/src/avm2/globals/flash/display/NativeWindow.as

```diff
@@ -0,0 +1,317 @@
+      _systemChrome = initOptions.systemChrome;
+      _transparent = initOptions.transparent;
+      _type = initOptions.type;
+    }
+
+    public function get width():Number
+    {
+      stub_getter("flash.display.NativeWindow", "width");
+      return _stage.stageWidth;
+    }
+
+    public function set width(value:Number):void
+    {
+      stub_setter("flash.display.NativeWindow", "width");
+      _stage.stageWidth = value;
+    }
+
+    public function get height():Number
+    {
+      stub_getter("flash.display.NativeWindow", "height");
+      return _stage.stageHeight;
+    }
+
+    public function set height(value:Number):void
+    {
+      stub_setter("flash.display.NativeWindow", "height");
+      _stage.stageHeight = value;
+    }
+
+    public function get x():Number
+    {
+      stub_getter("flash.display.NativeWindow", "x");
+      return _stage.x;
+    }
+
+    public function set x(value:Number):void
+    {
+      stub_setter("flash.display.NativeWindow", "x");
+    }
+
+    public function get y():Number
+    {
+      stub_getter("flash.display.NativeWindow", "y");
+      return _stage.y;
+    }
+
+    public function set y(value:Number):void
+    {
+      stub_setter("flash.display.NativeWindow", "y");
+    }
+
+    public function get bounds():Rectangle
+    {
+      stub_getter("flash.display.NativeWindow", "bounds");
+      return _bounds;
+    }
+
+    public function set bounds(value:Rectangle):void
+    {
+      stub_setter("flash.display.NativeWindow", "bounds");
+      _bounds.height = height = value.height;
+      _bounds.width = width = value.width;
+    }
+
+    public function get maximizable():Boolean
+    {
+      stub_getter("flash.display.NativeWindow", "maximizable");
+      return _maximizable;
+    }
+
+    public function get minimizable():Boolean
+    {
+      stub_getter("flash.display.NativeWindow", "minimizable");
+      return _minimizable;
+    }
+
+    public function get resizable():Boolean
+    {
+      stub_getter("flash.display.NativeWindow", "resizable");
+      return _resizable;
+    }
+
+    public function get systemChrome():String
+    {
+      stub_getter("flash.display.NativeWindow", "systemChrome");
+      return _systemChrome;
+    }
+
+    public function get transparent():Boolean
+    {
+      stub_getter("flash.display.NativeWindow", "transparent");
+      return _transparent;
+    }
+
+    public function get type():String
+    {
+      stub_getter("flash.display.NativeWindow", "type");
+      return _type;
+    }
+
+    public function get stage():Stage
+    {
+      return _stage;
+    }
+
+    // Activates this window.
+    public function activate():void
+    {
+      stub_method("flash.display.NativeWindow", "activate");
+      dispatchEvent(new Event(Event.ACTIVATE));
+    }
+
+    // Closes this window.
+    public function close():void
+    {
+      stub_method("flash.display.NativeWindow", "close");
+      if (!dispatchEvent(new Event(Event.CLOSING, false, true)))
```

Comment:
Isn't the negation wrong? I would think if it returns `false` we should not close?

Reply:
Yes indeed, I don’t remember at all why negation is here
```suggestion
      if (dispatchEvent(new Event(Event.CLOSING, false, true)))
```